File: python/anagram.py
#code
for t in range(int(input())):
    a = input()
    b = input()
    length = len(a)
    length1 = len(b)
    flag = 1
    count = [0]*256
    count1  = [0]*256
    if(length == length1):
        for i in a:
            count[ord(i)]+=1
            
        for j in b:
            count1[ord(j)]+=1
        
        for x in range(1,256):
            if(count[x]!=count1[x]):
                flag = 0
                break
        
    else:
        flag =0
    
    
    if(flag == 1):
        print("YES")
    if(flag ==0):
        print("NO")

File: anagram.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin>>t;
    while(t--){
        string str;
        string str1;
        cin>>str;
        cin>>str1;
        int a[256] = {0};
        int b[256] = {0};
        int l = str.length();
        int k = str1.length();
        for(int i = 0;i<l;i++){
            a[str[i]]++;
        }
        for(int j =0;j<k;j++){
            b[str1[j]]++;
        }
        int flag =1;
        for(int i=0;i<256;i++){
            if(a[i]==b[i]){
                continue;
            }
            else{
                flag =0;
                break;
            }
        }
        if(flag ==1){
            cout<<"YES"<<endl;
        }
        if(flag ==0){
            cout<<"NO"<<endl;
        }
    }
	//code
	return 0;
}
